jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/vsdm/VsdmReceiver.java

```diff
@@ -35,75 +37,130 @@
 	private static Coder coder = J2735.getPERUnalignedCoder();
 
 	private DatagramSocket socket;
-	
+
 	private OdeProperties odeProperties;
 
+	private SerializableMessageProducerPool<String, byte[]> messageProducerPool;
+
 	@Autowired
 	public VsdmReceiver(OdeProperties odeProps) {
-		
+
 		this.odeProperties = odeProps;
 
 		try {
-			socket = new DatagramSocket(odeProperties.getVsdmPort());
-			logger.info("[VSDM Receiver] Created UDP socket bound to port", odeProperties.getVsdmPort());
+			socket = new DatagramSocket(odeProperties.getReceiverPort());
+			logger.info("[VSDM Receiver] Created UDP socket bound to port ", odeProperties.getReceiverPort());
 		} catch (SocketException e) {
-			logger.error("[VSDM Receiver] Error creating socket with port ", odeProperties.getVsdmPort(), e);
+			logger.error("[VSDM Receiver] Error creating socket with port ", odeProperties.getReceiverPort(), e);
 		}
+
+		messageProducerPool = new SerializableMessageProducerPool<>(odeProperties);
 	}
 
 	@Override
 	public void run() {
 
-		while (true) {
-			
-			DatagramPacket packet = new DatagramPacket(new byte[odeProperties.getVsdmBufferSize()], odeProperties.getVsdmBufferSize());
+		logger.info("Vsdm Receiver Service started.");
+
+		byte[] buffer = new byte[odeProperties.getVsdmBufferSize()];
+
+		DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+
+		Boolean stopped = false;
+		while (!stopped) {
 
 			try {
+				logger.info("VSDM RECEIVER - Waiting for UDP packets...");
 				socket.receive(packet);
+				logger.info("VSDM RECEIVER - Packet received.");
+				String obuIp = packet.getAddress().getHostAddress();
+				int obuPort = packet.getPort();
+				logger.info("Packet length: {}, Buffer length: {}", packet.getLength(), buffer.length);
+				byte[] actualPacket = Arrays.copyOf(packet.getData(), packet.getLength());
+				if (packet.getLength() > 0) {
+					logger.info("VSDM RECEIVER - Received data:", buffer);
+					decodeData(actualPacket, obuIp, obuPort);
+				}
 			} catch (IOException e) {
-				logger.error("[VSDM Receiver] Error receiving UDP packet", e);
-			}
-
-			InputStream ins = new ByteArrayInputStream(packet.getData());
-			AbstractData decoded = null;
-			try {
-				decoded = coder.decode(ins, decoded);
-			} catch (DecodeFailedException | DecodeNotSupportedException e) {
-				logger.error("[VSDM Receiver] Error, unable to decode UDP message", e);
+				logger.error("VSDM RECEIVER - Error receiving UDP packet", e);
+				stopped = true;
```

Comment:
I see, I guess I was mixing up the old code with the new. Even so, this is a singleton thread and once it stops, it's gone forever until you restarting ODE.  So the only way to detect that the thread is gone would be to go back and find the instant that it stopped. Whereas, if you keep going, it is likely that you get a repeating error log.